File: client-mvvm/app/src/main/java/ua/meugen/android/rxjavatests_client/viewmodel/rest/RestApi.java
package ua.meugen.android.rxjavatests_client.viewmodel.rest;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;
import ua.meugen.android.rxjavatests_client.model.responses.DataResponse;

/**
 * @author meugen
 */

public interface RestApi {

    @GET("/rs/data/with/delay")
    Observable<DataResponse> dataWithDelay(
            @Query("delay") long delay,
            @Query("count") int count);
}

File: client-kotlin/app/src/main/java/ua/meugen/android/rxjavatests_client/presenter/rest/RestApi.kt
package ua.meugen.android.rxjavatests_client.presenter.rest

import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable
import ua.meugen.android.rxjavatests_client.model.DataResponse

/**
 * @author meugen
 */

interface RestApi {

    @GET("/rs/data/with/delay")
    fun dataWithDelay(
            @Query("delay") delay: Long,
            @Query("count") count: Int): Observable<DataResponse>
}
